src/ext/handlersToResp.c

```diff
@@ -188,6 +183,16 @@ static RdbRes toRespNewKey(RdbParser *p, void *userData, RdbBulk key, RdbKeyInfo
     ctx->keyCtx.keyLen = RDB_bulkLen(p, key);
     if ((ctx->keyCtx.key = RDB_bulkClone(p, key)) == NULL)
         return RDB_ERR_FAIL_ALLOC;
+
+    if (ctx->conf.delKeyBeforeWrite) {
+        struct iovec iov[4];
+        char keyLenStr[32];
+        IOV_CONST_STR(&iov[0], "*2\r\n$3\r\nDEL\r\n$");
+        iov_stringLen(&iov[1], ctx->keyCtx.keyLen, keyLenStr);
+        IOV_STRING(&iov[2], ctx->keyCtx.key, ctx->keyCtx.keyLen);
+        IOV_CONST_STR(&iov[3], "\r\n");
+        return writevWrap(ctx, iov, 4, 1, 1);
+    }
```

Comment:
maybe we can know the key type at this stage, and if we know it's gonna be handled by SET or RESTORE then we don't need that DEL?

Reply:
SET: We do pass the opcode in `RdbKeyInfo *info`. I can add an exception.
RESTORE: Known internally by the parser but not exposed via API (to librdb-extension).

Reply:
so since this is a very common use (of the tool, not the parser), and the DELs will make a huge overhead, maybe we wanna handle it?
or maybe we can just make sure that `delKeyBeforeWrite` is off when RESTORE will be used?

Reply:
> or maybe we can just make sure that delKeyBeforeWrite is off when RESTORE will be used?

Note that there is an option to multiple levels regsitration at different levels and also function `RDB_handleByLevel` that helps to resolve conflicts such that eventually the parser has a mapping per opcode of which level should be called.

Maybe instead of overloading with another getter function maybe to extend `RdbKeyInfo *info` to pass another argument:
`RdbHandlersLevel handledByLevel`.